packages/webpack/src/executors/webpack/lib/get-webpack-config.ts

```diff
@@ -56,43 +72,53 @@ export function getWebConfig(
   // TODO(jack): Replace merge behavior with an inlined config so it is easier to understand.
   return webpackMerge.merge([
     _getBaseWebpackPartial(
+      context,
       options,
       esm,
       isScriptOptimizeOn,
       tsConfig.options.emitDecoratorMetadata,
-      configuration
-    ),
-    getPolyfillsPartial(
-      options.polyfills,
-      options.es2015Polyfills,
-      esm,
-      isScriptOptimizeOn
-    ),
-    getStylesPartial(
-      wco.root,
-      wco.projectRoot,
-      wco.buildOptions,
-      options.extractCss,
-      options.postcssConfig
+      overrides
     ),
+    options.target === 'web'
+      ? getPolyfillsPartial(
+          options.polyfills,
+          options.es2015Polyfills,
+          esm,
+          isScriptOptimizeOn
+        )
+      : {},
+    options.target === 'web'
+      ? getStylesPartial(
+          wco.root,
+          wco.projectRoot,
+          wco.buildOptions,
+          options.extractCss,
+          options.postcssConfig
+        )
+      : {},
     getCommonPartial(wco),
-    getBrowserConfig(wco),
+    options.target === 'web' ? getBrowserConfig(wco) : {},
   ]);
 }
 
 function _getBaseWebpackPartial(
-  options: WebWebpackExecutorOptions,
+  context: ExecutorContext,
+  options: NormalizedWebpackExecutorOptions,
   esm: boolean,
   isScriptOptimizeOn: boolean,
   emitDecoratorMetadata: boolean,
-  configuration?: string
+  overrides?: GetWebpackConfigOverrides
 ) {
-  let partial = getBaseWebpackPartial(options, {
-    esm,
-    isScriptOptimizeOn,
-    emitDecoratorMetadata,
-    configuration,
-  });
+  let partial = getBaseWebpackPartial(
+    options,
+    {
+      esm,
+      isScriptOptimizeOn,
+      emitDecoratorMetadata,
+      configuration: overrides?.configuration ?? context.configurationName,
+    },
```

Comment:
Looks like the skipTypeCheck property is missing. How to override it from executor config?